core/domain/exp_services.py

```diff
@@ -1176,33 +1181,41 @@ def revert_exploration(
     else:
         exploration.validate()
 
-    exp_models.ExplorationModel.revert(
-        exploration_model, committer_id,
-        'Reverted exploration to version %s' % revert_to_version,
-        revert_to_version)
-    exp_memcache_key = exp_fetchers.get_exploration_memcache_key(exploration.id)
-    memcache_services.delete(exp_memcache_key)
-
-    # Update the exploration summary, but since this is just a revert do
-    # not add the committer of the revert to the list of contributors.
-    update_exploration_summary(exploration_id, None)
-
-    exploration_stats = stats_services.get_stats_for_new_exp_version(
-        exploration.id, current_version + 1, exploration.states,
-        exp_versions_diff=None, revert_to_version=revert_to_version)
-    stats_services.create_stats_model(exploration_stats)
-
-    current_exploration = exp_fetchers.get_exploration_by_id(
-        exploration_id, version=current_version)
-    stats_services.update_exp_issues_for_new_exp_version(
-        current_exploration, exp_versions_diff=None,
-        revert_to_version=revert_to_version)
-
-    if feconf.ENABLE_ML_CLASSIFIERS:
-        exploration_to_revert_to = exp_fetchers.get_exploration_by_id(
-            exploration_id, version=revert_to_version)
-        classifier_services.create_classifier_training_job_for_reverted_exploration( # pylint: disable=line-too-long
-            current_exploration, exploration_to_revert_to)
+    def update_storage_models():
+        """Commits all the newly created models in an atomic transaction. If any
+        operation fails, every other operation will fail as well.
+        """
+        exp_models.ExplorationModel.revert(
+            exploration_model, committer_id,
+            'Reverted exploration to version %s' % revert_to_version,
+            revert_to_version)
+
+        memcache_services.delete(
```

Comment:
Same question as above.

Reply:
Discussed offline -- proposed solution seems fine: only delete memcache when transaction succeeded, otherwise keep it around.